File: src/main/java/it/xpug/kata/birthday_greetings/MailSender.java
package it.xpug.kata.birthday_greetings;

import javax.mail.Message;
import javax.mail.MessagingException;
import javax.mail.Session;
import javax.mail.Transport;
import javax.mail.internet.AddressException;
import javax.mail.internet.InternetAddress;
import javax.mail.internet.MimeMessage;
import java.util.Properties;

public class MailSender {
    private String smtpHost;
    private int smtpPort;

    public MailSender(String smtpHost, int smtpPort) {
        this.smtpHost = smtpHost;
        this.smtpPort = smtpPort;
    }

    public void sendGreetingsTo(Employee employee) throws AddressException, MessagingException {
        String recipient = employee.getEmail();
        String body = "Happy Birthday, dear %NAME%!".replace("%NAME%", employee.getFirstName());
        String subject = "Happy Birthday!";
        String sender = "sender@here.com";

        // Create a mail session
        Properties props = new Properties();
        props.put("mail.smtp.host", smtpHost);
        props.put("mail.smtp.port", "" + smtpPort);
        Session session = Session.getInstance(props, null);

        // Construct the message
        Message msg = new MimeMessage(session);
        msg.setFrom(new InternetAddress(sender));
        msg.setRecipient(Message.RecipientType.TO, new InternetAddress(recipient));
        msg.setSubject(subject);
        msg.setText(body);

        // Send the message
        Transport.send(msg);
    }
}

File: src/main/java/it/xpug/kata/birthday_greetings/MailSender.kt
package it.xpug.kata.birthday_greetings

import java.util.*
import javax.mail.Message
import javax.mail.Session
import javax.mail.Transport
import javax.mail.internet.InternetAddress
import javax.mail.internet.MimeMessage

class MailSender(private val smtpHost: String, private val smtpPort: Int) : Sender {
    override fun send(sender: String, subject: String, body: String, recipient: String) {
        val session = Session.getInstance(properties(), null)

        Transport.send(MimeMessage(session).buildWith(sender, recipient, subject, body))
    }

    private fun properties(): Properties {
        return Properties().also {
            it["mail.smtp.host"] = smtpHost
            it["mail.smtp.port"] = "" + smtpPort
        }
    }
}
